rllib/agents/trainer.py

```diff
@@ -1878,12 +1871,8 @@ def log_result(self, result: ResultDict) -> None:
     @override(Trainable)
     def cleanup(self) -> None:
         # Stop all workers.
-        workers = getattr(self, "workers", None)
-        if workers:
-            workers.stop()
-        # Stop all optimizers.
-        if hasattr(self, "optimizer") and self.optimizer:
```

Comment:
why get rid of this?

Reply:
Seemed really outdated code.
Trainers do not have `self.optimizer` (anymore), only ES and ARS and those two optimizers do NOT have a `stop()` method, so this would actually produce errors here.

If users want their Trainers to have a `self.optimizer`, they can just override `Trainer.cleanup()` and implement the necessary logic.